source/routing/raptor_api.cpp

```diff
@@ -69,22 +71,29 @@ static void fill_shape(pbnavitia::Section* pb_section,
         // As every stop times may not be present (because they can be
         // filtered because of estimated datetime), we can only print
         // the shape if the 2 stop times are consecutive
-        if (prev_order + 1 == cur_order && st-> shape_from_prev != nullptr) {
-            for (const auto& cur_coord: *st->shape_from_prev) {
-                if (cur_coord == prev_coord) { continue; }
-                add_coord(cur_coord, pb_section);
-                prev_coord = cur_coord;
+        if (prev_order + 1 == cur_order) {
+            if (st-> shape_from_prev != nullptr) {
```

Comment:
There is an extra space after `->`.

It might be great to have a comment explaining what you're doing on this test, i.e. something like "if we don't have shape, use the intermediate stop points to generate the geometry"